### PR TITLE
[SITES-4716]: implement addComponentWithRetry() to fix add component …

### LINKED ISSUE #2025: Add component to a page fails
## Bug Report

**Current Behavior**
Add component to a page fails.

**Expected behavior/code**
Add component should be retried multiple times to add it after page creation.

**Environment**
- AEM version and patch level **df870e8857**
- Core Components version **2.18.0**

**Possible Solution**

**Additional context / Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

### PR DESCRIPTION
**Pull request by dhavkuma**
…failure due to page creation delay

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #2025 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

**Comment by dhavkuma** [comment]
@absarasw  Please review and approve the PR.

**Comment by vladbailescu** [comment]
@absarasw Since yesterday we switched to a single branch, named `main`

File: testing/it/e2e-selenium-utils/src/main/java/com/adobe/cq/wcm/core/components/it/seljup/util/Commons.java [hunk 1]
```diff
@@ -459,6 +460,47 @@ public static String addComponent(CQClient client, String component, String pare
...
+     * @param parentCompPath     mandatory absolute path to the parent component
+     * @param nameHint           optional hint for the component nodes name, if empty component name is taken
+     * @param order              optional where to place component e.g. 'before product_grid', if empty, 'last' is used
+     * @param timeout            try retries for this time duration
+     * @param delay              interval between successive retries
+     * @throws ClientException
+     */
+
+    public static String addComponentWithRetry(final CQClient client, final String component, final String parentCompPath, final String nameHint,
+                                               final String order, long timeout, long delay, final int... expectedStatus) throws ClientException {
```
Comment (vladbailescu) [inline testing/it/e2e-selenium-utils/src/main/java/com/adobe/cq/wcm/core/components/it/seljup/util/Commons.java:477]: `nameHint`, `order` and `expectedStatus` do not seem to be used

**Review by vladbailescu — APPROVED** [review]

File: testing/it/e2e-selenium-utils/src/main/java/com/adobe/cq/wcm/core/components/it/seljup/util/Commons.java [hunk 2]
```diff
@@ -459,6 +460,47 @@ public static String addComponent(CQClient client, String component, String pare
...
+
+        class AddComponentPolling extends Polling {
+            String componentPath;
+
+            AddComponentPolling() {
+            }
+
+            public Boolean call() throws Exception {
+                this.componentPath = addComponent(client, component, parentCompPath, "text", null);
```
Comment (absarasw) [inline testing/it/e2e-selenium-utils/src/main/java/com/adobe/cq/wcm/core/components/it/seljup/util/Commons.java:486]: I think it should be addComponent(client, component, parentCompPath, nameHint, order, expectedStatus);

cc: @vladbailescu

**Comment by dhavkuma** [comment]
Please review. @vladbailescu @absarasw.
Please merge if changes look fine. @vladbailescu

**Review by vladbailescu — APPROVED** [review]

**Pull request merged**